### PR TITLE
docs: update connection pooling docs

### PR DESCRIPTION
**Pull request by danieltprice**
- add a note about the 4 reserved connections for superuser
https://neon-next-git-dprice-update-connection-pool-2e11d1-neondatabase.vercel.app/docs/connect/connection-pooling#connection-limits-without-connection-pooling
- add a warning about the 64 per user/db pgBouncer pooler limit
https://neon-next-git-dprice-update-connection-pool-2e11d1-neondatabase.vercel.app/docs/connect/connection-pooling#connection-pooling
- move "How to use connection pooling" section to the top
- Other minor revisions, add table showing max_connections per compute size, update screen cap

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
image

Maybe a sidebar but "How do I use the 10,000 connections in my application?"

**Comment by rishi-raj-jain** [comment]
image

Both of the paragprahs mismatch each other in terms that I as a user is super confused how to create a new "user" that will allow me to increase the limit?

**Comment by rishi-raj-jain** [comment]
Also, can we add that to query the number of connectiosn open at a given time, one can run the following SQL statement:

```
SELECT usename FROM pg_stat_activity WHERE datname = 'your-main-database';
```

**Comment by rishi-raj-jain** [comment]
Also, can we include that queries if any running in the console would also decrease the cap of the number of connections that can be opened?

**Comment by danieltprice** [comment]
> Also, can we include that queries if any running in the console would also decrease the cap of the number of connections that can be opened?

For sure. @rishi-raj-jain Do you know if it is one connection that's used from the SQL Editor in the console?

**Comment by danieltprice** [comment]
> Both of the paragprahs leave me confused how to create a new "user" that will allow me to increase the limit?

@rishi-raj-jain Is this just a matter of "how to create a postgres user?" Do you think that a link like this would work:
For information about creating additional Postgres users, see x." or is there more to it? How would you handle this from an application perspective? If you have one connection string defined for your app and are running into the 64 connection limit, how would you fix it?

**Comment by danieltprice** [comment]
> Also, can we add that to query the number of connectiosn open at a given time, one can run the following SQL statement:
> 
> ```
> SELECT usename FROM pg_stat_activity WHERE datname = 'your-main-database';
> ```

Added.

**Comment by danieltprice** [comment]
> > Also, can we include that queries if any running in the console would also decrease the cap of the number of connections that can be opened?
> 
> For sure. @rishi-raj-jain Do you know if it is one connection that's used from the SQL Editor in the console?

Added that the SQL Editor uses a connect.

**Comment by danieltprice** [comment]
> Both of the paragprahs leave me confused how to create a new "user" that will allow me to increase the limit?

Added a link to the Manage roles section for create role instructions.

**Comment by danieltprice** [comment]
> Maybe a sidebar but "How do I use the 10,000 connections in my Node.js/Python/Rust application?"

Let's tackle this one later.

**Comment by rishi-raj-jain** [comment]
> > Also, can we include that queries if any running in the console would also decrease the cap of the number of connections that can be opened?
> 
> For sure. @rishi-raj-jain Do you know if it is one connection that's used from the SQL Editor in the console?

Yes.

**Comment by rishi-raj-jain** [comment]
> > Both of the paragprahs leave me confused how to create a new "user" that will allow me to increase the limit?
> 
> @rishi-raj-jain Is this just a matter of "how to create a postgres user?" Do you think that a link like this would work: For information about creating additional Postgres users, see x." or is there more to it? How would you handle this from an application perspective? If you have one connection string defined for your app and are running into the 64 connection limit, how would you fix it?

Exactly, that's what I want to learn myself.

File: content/docs/connect/connection-pooling.md [hunk 1]
```diff
@@ -7,45 +7,93 @@ redirectFrom:
...
 
-## Enable connection pooling
+Neon uses `PgBouncer` in `transaction mode`. For limitations associated with `transaction mode`, see [Connection pooling notes and limitations](#connection-pooling-notes). For more information about `PgBouncer`, refer to [https://www.pgbouncer.org/](https://www.pgbouncer.org/).
 
-Enabling connection pooling in Neon requires adding a `-pooler` suffix to the compute endpoint ID, which is part of the Neon hostname. Connections that specify the `-pooler` suffix in the connection string use a pooled connection.
+## Neon PgBouncer configuration settings
```
Comment (rishi-raj-jain) [inline content/docs/connect/connection-pooling.md:73]: may we add that one can contact support if they are on paid plan to have these changed for their requirements?

File: content/docs/connect/connection-pooling.md [hunk 2]
```diff
@@ -7,45 +7,93 @@ redirectFrom:
...
 
-You can add the `-pooler` suffix to the endpoint ID in your connection string as shown:
+Neon's PgBouncer configuration is shown below. The settings are not user-configurable.
```
Comment (rishi-raj-jain) [inline content/docs/connect/connection-pooling.md:75]: may we add that one can contact support if they are on paid plan to have these changed for their requirements?

**Review by rishi-raj-jain — CHANGES_REQUESTED** [review]

File: content/docs/connect/connection-pooling.md (hunk 2 above)
Comment (danieltprice) [reply content/docs/connect/connection-pooling.md]: added.

**Review by rishi-raj-jain — APPROVED** [review]

**Pull request merged**